src/Common/MoistThermodynamics/MoistThermodynamics.jl

```diff
@@ -827,20 +878,65 @@ The dry potential temperature where
 and, optionally,
  - `q` [`PhasePartition`](@ref). Without this argument the results are that of dry air.
  """
-dry_pottemp(T::FT, p::FT, q::PhasePartition{FT}=PhasePartition{FT}(FT(0), FT(0), FT(0))) where {FT<:Real} =
-  T / exner(p, q)
+dry_pottemp_given_pressure(T::FT, p::FT, q::PhasePartition{FT}=q_pt_0(FT)) where {FT<:Real} =
+  T / exner_given_pressure(p, q)
 
 """
     dry_pottemp(ts::ThermodynamicState)
 
 The dry potential temperature, given a thermodynamic state `ts`.
 """
 dry_pottemp(ts::ThermodynamicState) =
-  dry_pottemp(air_temperature(ts), air_pressure(ts), PhasePartition(ts))
+  dry_pottemp(air_temperature(ts), air_density(ts), PhasePartition(ts))
 dry_pottemp(ts::PhaseDry) = air_temperature(ts) / exner(ts)
 
 """
-    air_temperature_from_liquid_ice_pottemp(θ_liq_ice, p[, q::PhasePartition])
+    air_temperature_from_liquid_ice_pottemp(θ_liq_ice, ρ, q::PhasePartition)
+
+Air temperature
+``
+  T = θ_{liq-ice}*(ρ*R_m*θ_{liq-ice}/MSLP)^{R_m/c_{vm}} + latent_heat_liq_ice(q)/c_{vm}
+``
+given
+ - `θ_liq_ice` liquid-ice potential temperature
+ - `ρ` (moist-)air density
+and, optionally,
+ - `q` [`PhasePartition`](@ref). Without this argument the results are that of dry air.
+"""
+function air_temperature_from_liquid_ice_pottemp(θ_liq_ice::FT, ρ::FT,
+  q::PhasePartition{FT}=q_pt_0(FT)) where {FT<:Real}
+
+  R_m = gas_constant_air(q)
+  # Solve for temperature from liquid_ice_pottemp assuming no condensate:
+  T_dry = θ_liq_ice * (ρ*R_m*θ_liq_ice/FT(MSLP))^(R_m/cv_d)
+
+  # Now, add correction from condensate, assuming the relevant temperature is T_dry:
+  T_linearized = T_dry + latent_heat_liq_ice(q)/cv_d
```

Comment:
Same here: Replace `cv_d` by `cv_m`. Perhaps better to write the whole thing in one line 
```suggestion
  T_linearized = θ_liq_ice * (ρ*R_m*θ_liq_ice/FT(MSLP))^(R_m/cv_m) + latent_heat_liq_ice(q)/cv_d
```
(or rename `T_dry` to `T_unsaturated` because this is more precisely what it is).